savant/deepstream/utils/object.py

```diff
@@ -154,32 +139,147 @@ def nvds_generate_obj_uid(
             frame_meta.source_id,
             frame_meta.frame_num,
             obj_meta.obj_label,
-            bbox.x_center,
-            bbox.y_center,
+            bbox.xc,
+            bbox.yc,
             bbox.width,
             bbox.height,
         )
     )
 
 
-def nvds_get_obj_bbox(nvds_obj_meta: pyds.NvDsObjectMeta) -> Union[NvDsBBox, NvDsRBBox]:
-    """Returns BBox instance for specified frame meta.
+def nvds_get_obj_bbox(nvds_obj_meta: pyds.NvDsObjectMeta) -> Union[BBox, RBBox]:
+    """Get the bounding box for specified object meta
+    from Deepstream meta structures.
 
     :param nvds_obj_meta: NvDsObjectMeta.
-    :return:
+    :return: BBox or RBBox.
     """
-    if nvds_get_obj_selection_type(nvds_obj_meta) == ObjectSelectionType.REGULAR_BBOX:
-        return NvDsBBox(nvds_obj_meta)
+    bbox_type = nvds_get_obj_selection_type(nvds_obj_meta)
+
+    if bbox_type == ObjectSelectionType.REGULAR_BBOX:
+        return get_aligned_bbox_from_obj_meta(nvds_obj_meta)
 
-    if nvds_get_obj_selection_type(nvds_obj_meta) == ObjectSelectionType.ROTATED_BBOX:
-        return NvDsRBBox(nvds_obj_meta)
+    if bbox_type == ObjectSelectionType.ROTATED_BBOX:
+        return get_rotated_bbox_from_obj_meta(nvds_obj_meta)
 
     raise IncorrectSelectionType('Unsupported object selection type.')
 
 
+def nvds_set_obj_bbox(
+    batch_meta: pyds.NvDsBatchMeta,
+    obj_meta: pyds.NvDsObjectMeta,
+    bbox: Union[BBox, RBBox],
+) -> None:
+    """Set bbox values and object selection type for specified object meta
+    into Deepstream meta structures."""
+    if isinstance(bbox, BBox):
+        set_aligned_bbox_for_obj_meta(obj_meta, bbox)
+        nvds_set_obj_selection_type(
+            obj_meta,
+            ObjectSelectionType.REGULAR_BBOX,
+        )
+
+    elif isinstance(bbox, RBBox):
+        set_rotated_bbox_for_obj_meta(batch_meta, obj_meta, bbox)
+        nvds_set_obj_selection_type(
+            obj_meta,
+            ObjectSelectionType.ROTATED_BBOX,
+        )
+
+
+def set_aligned_bbox_for_obj_meta(obj_meta: pyds.NvDsObjectMeta, bbox: BBox) -> None:
```

Comment:
Bring to a single notation. There are set_, get_, and nvds_set.

Reply:
Don't understand the required changes.

Should I add `nvds_` prefix to all functions?

Reply:
1) Some functions start with "set", others with "update", but they do the same thing.
2) We gave names to all functions beginning with "nvds", which work with deepstream meta

Reply:
Refactored.